### PR TITLE
DEV-4970: Enable portal to filter contributions by RP

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
Filters contribution by RP in portal
#### Why are we doing this? How does it help us?
Only show contributions related to the specific donation page
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
no
#### Have automated unit tests been added? If not, why?
yes
#### How should this change be communicated to end users?
n/a
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-4970
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

File: spa/src/hooks/usePortalContributionList.ts [hunk 1]
```diff
@@ -91,12 +91,12 @@ async function fetchContributions(contributorId: number, queryParams?: string) {
  */
 export function usePortalContributionList(
   contributorId?: number,
-  queryParams?: { ordering: string; interval?: string }
+  queryParams?: { ordering: string; interval?: string; revenue_program?: number }
```
Comment (nrh-cklimas) [inline spa/src/hooks/usePortalContributionList.ts:94]: Let's move `revenue_program` into a separate arg to emphasize it's required--to me it is more like `contributorId` than `ordering`, for example.

File: spa/src/hooks/usePortalContributionList.test.ts [hunk 2]
```diff
@@ -70,40 +75,49 @@ describe('usePortalContributionList', () => {
     expect(axiosMock.history.get.length).toBe(0);
   });
 
-  describe('When contributor ID is defined', () => {
+  it("doesn't fetch contributions if RP ID is undefined", async () => {
+    hook(undefined, queryParams);
```
Comment (nrh-cklimas) [inline spa/src/hooks/usePortalContributionList.test.ts:79]: This looks like it's testing that contributions aren't fetched when `contributorId` isn't defined. I think we need two tests: one where `contributorId` is undefined and one where `revenue_program` is undefined.

File: spa/src/hooks/usePortal.tsx [hunk 3]
```diff
@@ -72,6 +76,7 @@ export default function usePortal() {
 
   return {
     page,
+    pageIsLoading,
```
Comment (nrh-cklimas) [inline spa/src/hooks/usePortal.tsx:79]: Is this implied by `pageIsFetched` being false?

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/hooks/usePortal.tsx (hunk 3 above)
Comment (GuiMend) [reply spa/src/hooks/usePortal.tsx:79]: I think we could do it that way. I just defaulted to check for the loading status
image

File: spa/src/hooks/usePortalContributionList.test.ts (hunk 2 above)
Comment (GuiMend) [reply spa/src/hooks/usePortalContributionList.test.ts]: [5406fcf](https://github.com/newsrevenuehub/rev-engine/pull/1566/commits/5406fcf5948062f5462720d546b5fd433810df0b)

File: spa/src/hooks/usePortal.tsx (hunk 3 above)
Comment (GuiMend) [reply spa/src/hooks/usePortal.tsx]: [5406fcf](https://github.com/newsrevenuehub/rev-engine/pull/1566/commits/5406fcf5948062f5462720d546b5fd433810df0b)

File: spa/src/hooks/usePortalContributionList.ts (hunk 1 above)
Comment (GuiMend) [reply spa/src/hooks/usePortalContributionList.ts]: [5406fcf](https://github.com/newsrevenuehub/rev-engine/pull/1566/commits/5406fcf5948062f5462720d546b5fd433810df0b)

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**